tap_jira/__init__.py

```diff
@@ -109,6 +105,13 @@ def sync():
     Context.state["currently_syncing"] = None
     singer.write_state(Context.state)
 
+def switch_pk_for_users_stream():
+    """
+    Update pk for users stream to key for on-prem Jira instance.
+    """
+    if Context.client.is_on_prem_instance:
+        # ALL_STREAMS contains `Users` object at index 7
+        streams_.ALL_STREAMS[7].pk_fields = ["key"]
```

Comment:
We should not be making the ordering of the streams in `ALL_STREAMS` matter. Just call `discover()` in the sync function to get the catalog instead of loading the schema in 2 different places in code.

Reply:
We were already calling discovery but not using generated catalog which we updated.